scilpy/image/datasets.py

```diff
@@ -160,40 +169,23 @@ def voxmm_to_value(self, x, y, z, origin):
             is of length 1, return a scalar value.
         """
         if self.interpolation is not None:
-            if not self.is_voxmm_in_bound(x, y, z, origin):
-                eps = float(1e-8)  # Epsilon to exclude upper borders
-                if origin == 'corner':
-                    x = max(0,
-                            min(self.voxres[0] * (self.dim[0] - eps), x))
-                    y = max(0,
-                            min(self.voxres[1] * (self.dim[1] - eps), y))
-                    z = max(0,
-                            min(self.voxres[2] * (self.dim[2] - eps), z))
-                elif origin == 'center':
-                    x = max(-self.voxres[0] / 2,
-                            min(self.voxres[0] * (self.dim[0] - 0.5 - eps), x))
-                    y = max(-self.voxres[1] / 2,
-                            min(self.voxres[1] * (self.dim[1] - 0.5 - eps), y))
-                    z = max(-self.voxres[2] / 2,
-                            min(self.voxres[2] * (self.dim[2] - 0.5 - eps), z))
-                else:
-                    raise ValueError("Origin should be 'center' or 'corner'.")
-
-            coord = np.array(self.voxmm_to_vox(x, y, z, origin),
-                             dtype=np.float64)
+            # Checking if out of bound.
+            x, y, z = self._vox_to_closest_coord(x, y, z, origin)
 
             # Interpolation: Using dipy's pyx methods. The doc can be found in
             # the file dipy.core.interpolation.pxd. Dipy works with origin
             # center.
+            coord = np.array((x, y, z), dtype=np.float64)
```

Comment:
do we really need the precision of a float64?

Reply:
I don't think I'm the one who decided that. @arnaudbore , any opinion?

Reply:
Oups.... I tried, and it fails.

```
    result = trilinear_interpolate4d(self.data, coord)
  File "dipy/core/interpolation.pyx", line 357, in dipy.core.interpolation.trilinear_interpolate4d
ValueError: Buffer dtype mismatch, expected 'double' but got 'float'
```
I will revert and add a comment.